### ISSUE TITLE
Rule Request: Redundant import

### ISSUE THREAD

**Issue by biboran** [realm/SwiftLint#2248]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

Redundant imports often lead to the confusion about the responsibilities of declarations within a file. I don't know any reasons for why anyone would want to have import declarations of the module they don't use.  

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

I didn't find any

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:
```swift
import UIKit
class Foo { }
```

Wouldn't trigger:
```swift
import UIKit
class Foo { let bar = UIView() }
```

3. Should the rule be configurable, if so what parameters should be configurable?

No

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Such rule should be enabled by default

**Comment by jpsim** [comment]
This would require SwiftLint to run in a "compiled mode" as I've discussed before.

**Comment by biboran** [comment]
@jpsim is it something you don’t consider for implementation?

**Comment by marcelofabri** [comment]
@jpsim can we close this as https://github.com/realm/SwiftLint/pull/2381 was merged?

**Comment by jpsim** [comment]
Nice find, yes! 8fd15b0b
